builders/interact/ethereum-api/precompiles/batch.md

```diff
@@ -0,0 +1,267 @@
+
+!!! note
+     Typically if you wanted to send the native currency to or through a contract, you would have to set the value within the overall transaction object and interact with a payable function. However, since the batch precompile interacts directly with Substrate code, this is not a typical Ethereum transaction and is thus not necessary.
+
+### Find a Contract Interaction's Call Data {: #find-a-contract-interactions-call-data }
+
+Visual interfaces like [Remix](/builders/interact/ethereum-api/dev-env/remix/){target=\_blank} and handy libraries like [Ethers.js](/builders/interact/ethereum-api/libraries/ethersjs/){target=\_blank} hide the way that Ethereum transactions interact with Solidity smart contracts. The name and input types of a function are hashed into a [function selector](https://docs.soliditylang.org/en/latest/abi-spec.html#function-selector-and-argument-encoding){target=\_blank} and the input data is encoded. These two pieces are then combined and sent as the transaction's call data. To send a subtransaction within a batch transaction, the sender needs to know its call data beforehand.
+
+Try finding a transaction's call data using Remix:
+
+1. Expand the `SimpleContract.sol` contract under **Deployed Contracts**
+2. Expand the **setMessage** function
+3. Enter the input of the function. For this example, **id** will be `1` and **message** will be `"tanssi"`
+4. Instead of sending the transaction, click the copy button next to the **transact** button to copy the call data
+
+![Transaction Call Data](/images/builders/interact/ethereum-api/precompiles/batch/batch-5.webp)
+
+Now you have the transaction's call data! Considering the example values of `1` and `"tanssi"`, we can keep an eye out for their encoded values in the call data:
+
+```text
+0x648345c8                                                        // function selector
+0000000000000000000000000000000000000000000000000000000000000001  // 1 id
+0000000000000000000000000000000000000000000000000000000000000040  // 32 byte offset
+000000000000000000000000000000000000000000000000000000000000000   // 32 byte length
+674616e7373690000000000000000000000000000000000000000000000000000 // "tanssi" in bytes
+```
+
+The call data can be broken into five lines, where:
+
+ - The first line is the function selector
+ - The second line is equal to 1, which is the **id** that was provided
+ - What's left has to do with the **message** input. These last three lines are tricky since strings are a [dynamic type](https://docs.soliditylang.org/en/v0.8.15/abi-spec.html#use-of-dynamic-types){target=\_blank} with a dynamic length. The third line refers to an offset to define where the string's data starts. The fourth line refers to the length of the message in the following line, which is 32 bytes total - the "tanssi" message plus padding.  
```

Comment:
```suggestion
 - What's left has to do with the **message** input. These last three lines are tricky since strings are a [dynamic type](https://docs.soliditylang.org/en/v0.8.15/abi-spec.html#use-of-dynamic-types){target=\_blank} with a dynamic length. The third line refers to an offset to define where the string's data starts. The fourth line refers to the length of the message in the following line, which is 32 bytes total - the "tanssi" message plus padding
```